### PR TITLE
Design patterns p4

### PR DESCRIPTION
**Pull request by BorysovskaOA**

### CODE REVIEW

File: design_patterns/src/shipments/ShipmentInterface.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export interface ShipmentInterface {
```
Comment (AlexHladin) [inline design_patterns/src/shipments/ShipmentInterface.ts:1]: @BorysovskaOA, this interface volatiles the interface segregation principle. The `ship` method should be extracted into a separated interface.

File: design_patterns/src/shipments/ShipmentMarksDecorator.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import { Shipment } from './Shipment';
+import { ShipmentInterface } from './ShipmentInterface';
+
+export class ShipmentMarksDecorator implements ShipmentInterface{
+  constructor(private wrappee: Shipment, private marks: string[]) {
+  }
+
+  public ship() {
```
Comment (AlexHladin) [inline design_patterns/src/shipments/ShipmentMarksDecorator.ts:8]: @BorysovskaOA, decorate ship method only.

**Review by AlexHladin — CHANGES_REQUESTED** [review]

File: design_patterns/src/shipments/ShipmentInterface.ts (hunk 1 above)
Comment (BorysovskaOA) [reply design_patterns/src/shipments/ShipmentInterface.ts]: Created Shippable interface.

File: design_patterns/src/shipments/ShipmentMarksDecorator.ts (hunk 2 above)
Comment (BorysovskaOA) [reply design_patterns/src/shipments/ShipmentMarksDecorator.ts:7]: Removed extra methods from decorator.

**Review by AlexHladin — APPROVED** [review]

**Pull request merged**